src/overlays/actors/ovl_Dm_Hina/z_dm_hina.c

```diff
@@ -33,24 +33,156 @@ ActorInit Dm_Hina_InitVars = {
     (ActorFunc)DmHina_Draw,
 };
 
-#endif
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/DmHina_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/DmHina_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F470.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F56C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F5AC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F63C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F75C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/DmHina_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/func_80A1F9AC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Hina/DmHina_Draw.s")
+void DmHina_Init(Actor* thisx, PlayState* play) {
+    DmHina* this = THIS;
+
+    this->unk180 = 1;
+    this->actionFunc = func_80A1F470;
+    this->unk158 = this->actor.world.pos.y;
+    this->unk148 = 0.0f;
+    this->unk15C = 1.0f;
+    this->actor.focus.pos.x = this->actor.world.pos.x;
+    this->actor.focus.pos.y = this->actor.world.pos.y;
+    this->actor.focus.pos.z = this->actor.world.pos.z;
+}
+
+void DmHina_Destroy(Actor* thisx, PlayState* play) {
+}
+
+void func_80A1F470(DmHina* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    Math_SmoothStepToF(&this->unk148, 0.6f, 0.5f, 0.05f, 0.001f);
+    this->unk154 = Math_SinS(play->gameplayFrames * 0x708) * 8.0f;
+    if ((player->stateFlags1 & 0x400) && (this->actor.xzDistToPlayer < 80.0f)) {
+        this->unk180 = 0;
+        this->unk154 = 0.0f;
+        this->actor.world.pos.y += 40.0f;
+        this->actionFunc = func_80A1F56C;
+    }
+}
+
+void func_80A1F56C(DmHina* this, PlayState* play) {
+    if (Message_GetState(&play->msgCtx) == TEXT_STATE_CLOSING) {
+        this->unk17C = 2;
+        this->actionFunc = func_80A1F5AC;
+    }
+}
+
+void func_80A1F5AC(DmHina* this, PlayState* play) {
+    this->unk17C--;
+    if (this->unk17C == 0) {
+        this->unk180 = 1;
+        Cutscene_StartManual(play, &play->csCtx);
+        this->unk17A = Play_CreateSubCamera(play);
+        Play_ChangeCameraStatus(play, CAM_ID_MAIN, CAM_STATUS_WAIT);
+        Play_ChangeCameraStatus(play, this->unk17A, CAM_STATUS_ACTIVE);
+        this->actionFunc = func_80A1F63C;
+    }
+}
+
+void func_80A1F63C(DmHina* this, PlayState* play) {
+    this->unk160.x = this->actor.world.pos.x + 100.0f;
+    this->unk160.y = this->unk158 + 260.0f;
+    this->unk160.z = this->actor.world.pos.z + 100.0f;
+    this->unk16C.x = this->actor.world.pos.x;
+    this->unk16C.y = this->actor.world.pos.y + this->unk154 * this->unk15C + 40.0f * this->unk15C;
+    this->unk16C.z = this->actor.world.pos.z;
+    Play_SetCameraAtEye(play, this->unk17A, &this->unk16C, &this->unk160);
```

Comment:
```suggestion
    Play_SetCameraAtEye(play, this->subCamId, &this->subCamAt, &this->subCamEye);
```
This has a standardized name, could you name these as such?